pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Replicator.java

```diff
@@ -29,9 +29,9 @@ public interface Replicator {
 
     ReplicatorStatsImpl getStats();
 
-    CompletableFuture<Void> disconnect();
+    CompletableFuture<Void> terminate();
 
-    CompletableFuture<Void> disconnect(boolean b);
+    CompletableFuture<Void> terminate(boolean b);
```

Comment:
Sorry, I want to say can we have a new method terminate()? without touching the existing disconnect() method.

- disconnect(), you can reconnect later.
- terminate(), you can't restart on the same broker

Reply:
> `terminate`: you can't restart on the same broker

For now, all the scenarios calling `disconnect` want to close the replicator 😂 , so I modify them to call `terminate.`

> `disconnect`: you can reconnect later.

The method `closeProducerAsync` did the same thing, it only be used in the Replicator internal.